libraries/rush-lib/src/cli/actions/InstallAction.ts

```diff
@@ -49,13 +50,29 @@ export class InstallAction extends BaseInstallAction {
       parameterLongName: '--check-only',
       description: `Only check the validity of the shrinkwrap file without performing an install.`
     });
+
+    this._ignoreScriptsParameter = this.defineFlagParameter({
+      parameterLongName: '--ignore-scripts',
+      description:
+        'Do not execute any install lifecycle scripts specified in package.json files and its' +
+        ' dependencies when "rush install". Running with this flag leaves your installation in a uncompleted' +
+        ' state, you need to run without this flag again to complete a full installation. Meanwhile, it makes' +
+        ' your installing faster. Later, you can run "rush install" to run all ignored scripts. Moreover, you' +
+        ' can partial install such as "rush install --to <package>" to run ignored scripts of the dependencies' +
+        ' of the selected projects.'
+    });
```

Comment:
Is this only supported when using PNPM? We should either explicitly mention that and error if someone uses this flag in a non-pnpm repo, or just not include this flag at all in non-pnpm repos.

Reply:
> just not include this flag at all in non-pnpm repos.

Better to include the flag and throw an error saying `the "--ignore-scripts" parameter is not supported with your selected package manager`.

It is an infuriating user experience for CLI parameters to mysteriously disappear based on context. (*"I SWEAR Rush has this feature! Did I forget the name of it? Am I losing my mind?!"*  *"No seriously, look, the docs even mention it!"*  *"Did they remove the feature? Why would they do that?!"*) 😆